### PR TITLE
Group historical events according to reference. And order, newest first

### PR DESCRIPTION
**Pull request by edavey**
## Changes in this PR

- Group historical events by reference (and timestamp)
- Order, newest first

### Before
Ordered by creation time, ungrouped:

historical_events_ungrouped

### After
Ordered (newest first), and grouped by reference:

sorted_and_grouped

## Next steps

- [ ] Is an ADR required? An ADR should be added if this PR introduces a change to the architecture.
- [x] Is a changelog entry required? An entry should always be made in `CHANGELOG.md`, unless this PR is a small tweak which has no impact outside the development team.
- [ ] Do any environment variables need amending or adding?
- [ ] Have any changes to the XML been checked with the IATI validator? See [XML Validation](https://github.com/UKGovernmentBEIS/beis-report-official-development-assistance/blob/develop/doc/xml-validation.md)

### CODE REVIEW

**Comment by edavey** [comment]
[Group historical events by reference](https://trello.com/c/4JvqD9Lv/1996-group-historical-events-by-reference)

File: spec/controllers/staff/activity_historical_events_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  let(:organisation) { create(:delivery_partner_organisation) }
+
+  before do
+    allow(controller).to receive(:current_user).and_return(user)
+    allow(controller).to receive(:logged_in_using_omniauth?).and_return(true)
+  end
+
+  describe "#show" do
+    let(:activity) { build_stubbed(:project_activity) }
+    let(:grouper) { instance_double(Activity::HistoricalEventsGrouper, call: double)}
```
Comment (pezholio) [inline spec/controllers/staff/activity_historical_events_controller_spec.rb:14]: StandardRB doens't like this line

```suggestion
    let(:grouper) { instance_double(Activity::HistoricalEventsGrouper, call: double) }
```

File: app/assets/stylesheets/partials/_historical_events.css.scss [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    .govuk-summary-list__row {
+      dd, dt {
+        padding-top: 0;
+
+      }
+    }
+  }
+  table {
+
+  }
```
Comment (pezholio) [inline app/assets/stylesheets/partials/_historical_events.css.scss:15]: Do we need this here?

**Review by pezholio — APPROVED** [review]
LGTM - Just one teensy CSS thing, and also the build is broken due to StandardRb things

File: app/assets/stylesheets/partials/_historical_events.css.scss (hunk 2 above)
Comment (edavey) [reply app/assets/stylesheets/partials/_historical_events.css.scss:15]: No! well spotted, thks

File: spec/controllers/staff/activity_historical_events_controller_spec.rb (hunk 1 above)
Comment (edavey) [reply spec/controllers/staff/activity_historical_events_controller_spec.rb:14]: thks, should have noticed that one

**Pull request merged**